userdev/src/functionalTest/groovy/net/neoforged/gradle/userdev/FunctionalTests.groovy

```diff
@@ -155,6 +193,74 @@ class FunctionalTests extends BuilderBasedTestSpecification {
         and:
         def secondRun = project.run {
             it.tasks('build')
+            it.stacktrace()
+        }
+
+        then:
+        secondRun.task(":neoFormRecompile").outcome == TaskOutcome.FROM_CACHE
+        initialRun.task(":build").outcome == TaskOutcome.SUCCESS
+    }
+
+    def "the userdev runtime supports restricted repositories"() {
+        given:
+        def project = create("userdev_supports_loading_from_buildcache", {
+            it.build("""
+            java {
+                toolchain {
+                    languageVersion = JavaLanguageVersion.of(21)
+                }
+            }
+            
+            //Note: This cannot be static so that fg.repository can be properly accessed
```

Comment:
This comment you copy pasted is no longer valid (and I forgot to remove it when switching from FG to NG). Also I think instead of this helper method as you only have one repo, you probably could get away with:
```groovy
repositories {
    exclusiveContent {
        forRepository {
            maven { url 'https://maven.tterrag.com/' }
        }
        filter { includeGroup('team.chisel.ctm') }
    }
}
```